src/services/query-builder.js

```diff
@@ -32,21 +32,26 @@ function QueryBuilder(model, opts, params) {
     Object.values(modelForIncludes.associations)
       .filter((association) => ['HasOne', 'BelongsTo'].includes(association.associationType))
       .forEach((association) => {
+        const targetFields = Object.values(association.target.tableAttributes)
+          .map((attribute) => attribute.field);
+
         const explicitAttributes = (fieldNamesRequested || [])
           .filter((name) => name.startsWith(`${association.as}.`))
-          .map((name) => name.replace(`${association.as}.`, ''));
+          .map((name) => name.replace(`${association.as}.`, ''))
+          .filter((fieldName) => targetFields.includes(fieldName));
```

Comment:
If I'm not mistaken, you could use a single `reduce` here for clarity.
```
const explicitAttributes = (fieldNamesRequested || [])
	.reduce((result, name) => [
		...result,
		...(name.startsWith(`${association.as}.`) && targetFields.includes(name))
		? [name.replace(`${association.as}.`, '')] 
		: []
	]
, []);
```

Reply:
Still, you could regroup the two filters in a single one no ?

Reply:
@DrRaider nope, because the second filter is done with the value on which we removed the `${association.as}.` prefix, which is exactly what I want to get, to apply the filter correctly